app/components/course_preview/missing_information_component.rb

```diff
@@ -36,9 +40,14 @@ def train_with_disability_link = "#{about_publish_provider_recruitment_cycle_pat
     def train_with_us_link = "#{about_publish_provider_recruitment_cycle_path(provider_code, recruitment_cycle_year, course_code:, goto_preview: true)}#train-with-us"
 
     def about_accrediting_provider_link
-      return "#{edit_publish_provider_recruitment_cycle_accredited_provider_path(provider_code, recruitment_cycle_year, accredited_provider_code: @course.accredited_provider_code, goto_preview: true)}#accredited-provider-form-description-field" if FeatureService.enabled?(:accredited_provider_search)
-
-      "#{about_publish_provider_recruitment_cycle_path(provider_code, recruitment_cycle_year, course_code:, goto_preview: true)}#accrediting-provider-#{accrediting_provider.provider_code}"
+      if FeatureService.enabled?(:accredited_provider_search) && accrediting_provider_present?(course)
+        "#{edit_publish_provider_recruitment_cycle_accredited_provider_path(provider_code, recruitment_cycle_year, accredited_provider_code: @course.accredited_provider_code, goto_preview: true)}#accredited-provider-form-description-field"
```

Comment:
You can use the `anchor` option here instead of interpolating the whole URL. Similar for above and below.

```suggestion
        edit_publish_provider_recruitment_cycle_accredited_provider_path(provider_code, recruitment_cycle_year, accredited_provider_code: @course.accredited_provider_code, goto_preview: true, anchor: 'accredited-provider-form-description-field')
```

Reply:
Nice one, that's a lot better! 

https://github.com/DFE-Digital/publish-teacher-training/pull/3746/commits/5e68903d18b7865680824cb8abfdaf429c8eaa37